relay-profiling/src/android.rs

```diff
@@ -141,83 +133,23 @@ pub fn expand_android_profile(payload: &[u8]) -> Result<Vec<Vec<u8>>, ProfileErr
-
-        new_profile.profile_id = EventId::new();
-        new_profile.set_transaction(transaction);
-        new_profile.transactions.clear();
-
-        let clock = new_profile.profile.clock;
-        let start_time = new_profile.profile.start_time;
-
-        new_profile.profile.events.retain_mut(|event| {
-            let event_timestamp = get_timestamp(clock, start_time, event.time);
-            if transaction.relative_start_ns <= event_timestamp
-                && event_timestamp <= transaction.relative_end_ns
-            {
-                let relative_start = match clock {
-                    Clock::Cpu => Duration::from_millis(transaction.relative_cpu_start_ms),
-                    Clock::Wall | Clock::Dual | Clock::Global => {
-                        Duration::from_nanos(transaction.relative_start_ns)
-                    }
-                };
-                event.time = substract_to_timestamp(event.time, relative_start);
-                true
-            } else {
-                false
-            }
-        });
+    let transaction = &profile.transactions[0];
 
-        if new_profile.profile.events.is_empty() {
-            continue;
-        }
+    profile.transaction_name = transaction.name.clone();
+    profile.transaction_id = transaction.id;
+    profile.trace_id = transaction.trace_id;
+    profile.duration_ns = transaction.duration_ns();
 
-        match serde_json::to_vec(&new_profile) {
-            Ok(payload) => items.push(payload),
-            Err(_) => {
-                return Err(ProfileError::CannotSerializePayload);
-            }
-        };
-    }
+    profile.transactions.clear();
 
-    Ok(items)
-}
+    match serde_json::to_vec(&profile) {
+        Ok(payload) => items.push(payload),
+        Err(_) => {
+            return Err(ProfileError::CannotSerializePayload);
+        }
+    };
 
-fn substract_to_timestamp(time: Time, duration: Duration) -> Time {
-    match time {
-        Time::Global(time) => Time::Global(time - duration),
-        Time::Monotonic {
-            cpu: Some(time),
-            wall: None,
-        } => Time::Monotonic {
-            cpu: time
-                .checked_sub(duration)
-                .or_else(|| Some(Duration::default())),
-            wall: None,
-        },
-        Time::Monotonic {
-            wall: Some(time),
-            cpu: None,
-        } => Time::Monotonic {
-            wall: time
-                .checked_sub(duration)
-                .or_else(|| Some(Duration::default())),
-            cpu: None,
-        },
-        Time::Monotonic {
-            cpu: Some(cpu),
-            wall: Some(wall),
-        } => Time::Monotonic {
-            cpu: cpu
-                .checked_sub(duration)
-                .or_else(|| Some(Duration::default())),
-            wall: wall
-                .checked_sub(duration)
-                .or_else(|| Some(Duration::default())),
-        },
-        _ => unimplemented!(),
-    }
+    Ok(items)
```

Comment:
This function now always returns a vector with exactly one element. This can be simplified.

Reply:
Since other functions are still expanding, I need to keep the vector in the end, so I left it that way so I don't have to rewrite more of the existing logic.